### PR TITLE
Suppress some "'noreturn' function does return" warnings

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: arch/arm/src/common/up_pthread_start.c [hunk 1]
```diff
@@ -80,6 +80,8 @@ void up_pthread_start(pthread_startroutine_t entrypt, pthread_addr_t arg)
   /* Let sys_call2() do all of the work */
 
   sys_call2(SYS_pthread_start, (uintptr_t)entrypt, (uintptr_t)arg);
+
+  PANIC();
```
Comment (xiaoxiang781216) [inline arch/arm/src/common/up_pthread_start.c]: But don't see who declare up_pthread_start/up_task_start as noreturn_function

Comment (yamt) [reply arch/arm/src/common/up_pthread_start.c]: include/nuttx/arch.h has their prototypes with noreturn_function.

Comment (xiaoxiang781216) [reply arch/arm/src/common/up_pthread_start.c]: If so, we need to change all location to ensure the warning get clear for all arch

File: tools/mksyscall.c [hunk 2]
```diff
@@ -228,6 +228,11 @@ static void generate_proxy(int nparms)
   fprintf(stream, "/* Auto-generated %s proxy file -- do not edit */\n\n", g_parm[NAME_INDEX]);
   fprintf(stream, "#include <nuttx/config.h>\n");
 
+  /* Suppress "'noreturn' function does return" warnings. */
```
Comment (Ouss4) [inline tools/mksyscall.c]:
```suggestion
  /* Suppress "'noreturn' function does return" warnings. */
  
```

Comment (yamt) [reply tools/mksyscall.c]: done

File: arch/arm/src/common/up_pthread_start.c (hunk 1 above)
Comment (yamt) [reply arch/arm/src/common/up_pthread_start.c]: done

**Pull request merged**